app/forms/user_address_form.rb

```diff
@@ -0,0 +1,24 @@
+class UserAddressForm < AddressForm
+  attr_accessor :name, :phone, :user, :user_info, :user_address
+
+  validates :name, presence: true
+  validates :phone, presence: true
+
+  def save 
+    return false if invalid?
+    @user_info = create_user_info
+    @user_address = create_address(@user_info)
```

Comment:
same comment as shop_registration_form

Reply:
Updated:
- Instance variables are changed to local variables, except for `@user_address` as I need to access it when rendering new address
- #save now applies Transactions to create records